pkg/utils/utils_test.go

```diff
@@ -0,0 +1,262 @@
+package utils_test
+
+import (
+	multiv1beta1 "github.com/k8snetworkplumbingwg/multi-networkpolicy/pkg/apis/k8s.cni.cncf.io/v1beta1"
+	netdefv1 "github.com/k8snetworkplumbingwg/network-attachment-definition-client/pkg/apis/k8s.cni.cncf.io/v1"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"net"
+	"os"
+
+	"github.com/Mellanox/multi-networkpolicy-tc/pkg/utils"
+)
+
+var _ = Describe("utils test", func() {
+	Context("CheckNodeNameIdentical()", func() {
+		It("check node name identical with domain", func() {
+			h1 := "my-host.some-lab.some-site"
+			h2 := "my-host.some-lab-alt-name.some-site-alt-name"
+			Expect(utils.CheckNodeNameIdentical(h1, h2)).To(BeTrue())
+		})
+		It("check node name identical with domain on first arg", func() {
+			h1 := "my-host.some-lab.some-site"
+			h2 := "my-host"
+			Expect(utils.CheckNodeNameIdentical(h1, h2)).To(BeTrue())
+		})
+		It("check node name identical with domain on second arg", func() {
+			h1 := "my-host.some-lab.some-site"
+			h2 := "my-host"
+			Expect(utils.CheckNodeNameIdentical(h1, h2)).To(BeTrue())
+		})
+		It("check node name identical with same domain", func() {
+			h1 := "my-host.some-lab.some-site"
+			h2 := "my-host.some-lab.some-site"
+			Expect(utils.CheckNodeNameIdentical(h1, h2)).To(BeTrue())
+		})
+		It("check node name identical without domain", func() {
+			h1 := "my-host"
+			h2 := "my-host"
+			Expect(utils.CheckNodeNameIdentical(h1, h2)).To(BeTrue())
+		})
+		It("check node name not identical with domain", func() {
+			h1 := "my-host.some-lab.some-site"
+			h2 := "my-other-host.some-lab.some-site"
+			Expect(utils.CheckNodeNameIdentical(h1, h2)).To(BeFalse())
+		})
+		It("check node name not identical without domain", func() {
+			h1 := "my-host"
+			h2 := "my-other-host"
+			Expect(utils.CheckNodeNameIdentical(h1, h2)).To(BeFalse())
+		})
+	})
+
+	Context("GetHostname()", func() {
+		It("Gets hostname no override", func() {
+			host, err := os.Hostname()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(utils.GetHostname("")).To(Equal(host))
+		})
+		It("Gets hostname with override", func() {
+			dummyHost := "my-dummy-host"
+			Expect(utils.GetHostname(dummyHost)).To(Equal(dummyHost))
+		})
+	})
+
+	Context("IsMultiNetworkpolicyTarget()", func() {
+		createPodFn := func(phase v1.PodPhase, hostNetwork bool) *v1.Pod {
+			return &v1.Pod{
+				TypeMeta:   metav1.TypeMeta{},
+				ObjectMeta: metav1.ObjectMeta{},
+				Spec: v1.PodSpec{
+					HostNetwork: hostNetwork,
+				},
+				Status: v1.PodStatus{
+					Phase: phase,
+				},
+			}
+		}
+
+		It("returns true if pod is running and is not host network", func() {
+			pod := createPodFn(v1.PodRunning, false)
+			Expect(utils.IsMultiNetworkpolicyTarget(pod)).To(BeTrue())
+		})
+		It("returns false if pod is not running", func() {
+			pod := createPodFn(v1.PodPending, false)
+			Expect(utils.IsMultiNetworkpolicyTarget(pod)).To(BeFalse())
```

Comment:
nit: in utils package, IsMultiNetworkpolicyTarget can be converted to oneliner

Reply:
Done